virtcontainers/kata_agent.go

```diff
@@ -940,6 +945,50 @@ func (k *kataAgent) check() error {
 	return err
 }
 
+type reqFunc func(context.Context, interface{}, ...golangGrpc.CallOption) (interface{}, error)
+
+func (k *kataAgent) installReqFunc(c *kataclient.AgentClient) {
+	k.reqHandlers = make(map[string]reqFunc)
```

Comment:
Yes, in this case, I think the map is the only way to avoid the gocyclo complexity going over 15.
Any reason you didn't use a static map here ?

Reply:
Because the agent client is dynamically allocated. I also wanted to use a static map but it turns out there is no static grpc request handler.